agent/runtime-attach/src/main/java/com/microsoft/applicationinsights/attach/ApplicationInsights.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.attach;
+
+import io.opentelemetry.contrib.attach.RuntimeAttach;
+import java.io.File;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Optional;
+
+/** This class allows you to attach the Application Insights agent for Java at runtime. */
+public class ApplicationInsights {
+
+  public static final String RUNTIME_ATTACHED_PROPERTY = "agent.runtime.attached";
+
+  private ApplicationInsights() {}
+
+  /**
+   * Attach the Application Insights agent for Java to the current JVM. The attachment must be
+   * requested at the beginning of the main method.
+   */
+  public static void attach() {
+
+    if (!agentIsAttached()) {
+      System.setProperty(RUNTIME_ATTACHED_PROPERTY, "true");
+    }
+
+    Optional<File> optionalConfigFile = searchConfigFile();
+    if (optionalConfigFile.isPresent()) {
+      File configFile = optionalConfigFile.get();
+      System.setProperty("applicationinsights.configuration.file", configFile.getAbsolutePath());
+    }
+
+    RuntimeAttach.attachJavaagentToCurrentJVM();
+
+    if (!agentIsAttached()) {
+      System.setProperty(RUNTIME_ATTACHED_PROPERTY, "false");
```

Comment:
can you explain why line 45 and then setting it to false here?

Reply:
At this step, if the runtime attachment is not done, the `agent.runtime.attached` property always has a `true` value. So, for consistency it is set to `false`. However, since the agent is not attached, the property value won't be used.... So, I will remove this piece of code. The code will be easier to read without apparently any impact.